### ISSUE TITLE
Update vector rnets for schools

### ISSUE THREAD

**Issue by Robinlovelace** [npct/pct-shiny#676]
In a format that works with the current set-up, reproducible example to show what I mean:

``` r
z = readRDS("/home/rstudio/data/npct/pct-outputs-regional-R/school/lsoa/west-yorkshire/rnet.Rds")
class(z)
#> [1] "SpatialLinesDataFrame"
#> attr(,"package")
#> [1] "sp"
names(z)
#> Loading required package: sp
#> [1] "local_id"      "bicycle"       "govtarget_slc" "dutch_slc"
format(object.size(z), units = "MB")
#> [1] "31.6 Mb"
plot(z)
```

![](https://i.imgur.com/vZ7fryY.png)

<sup>Created on 2019-02-25 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by Robinlovelace** [comment]
Note: 31 MB is already on the big size. May need to simplify.

**Comment by mem48** [comment]
Robin is this a conversion of  my rnet https://github.com/mem48/pct-raster-tests/releases/tag/1.0.1 ?

If so look for lines where there are 0 cyclists in all 3 scenarios and remove them. I forgot to do this before producing the rnet. I'll update the file today

**Comment by Robinlovelace** [comment]
> I'll update the file today

Thanks @mem48 just to clarify: which file will you update?

I think the next stage will be to create an `rnet.Rds` file for each region an put them in the appropriate places:

- https://github.com/npct/pct-outputs-regional-R/tree/master/school/lsoa
- https://github.com/npct/pct-outputs-regional-notR/tree/master/school/lsoa

Sound good?

**Comment by mem48** [comment]
@Robinlovelace 

I've updated the national schools rnet https://github.com/mem48/pct-raster-tests/releases/tag/1.0.1 removing the lines with zero cyclists in all scenarios. Which reduced the file from 153 MB to 121 MB. I'd expect simialr savings at the regional level.

Are we having both raster an vector rnets for the schools layer in the website?

**Comment by Robinlovelace** [comment]
Great work @mem48 !

> Are we having both raster an vector rnets for the schools layer in the website?

Yes, that's the plan. Are you up for breaking them up into regional clusters and saving each in the relevant folder?

**Comment by Robinlovelace** [comment]
Almost done on this. Sanity check below. Right side is new - clearly better IMO:

![image](https://user-images.githubusercontent.com/1825120/53576042-bf166980-3b6a-11e9-9bbd-448e0ae09e2b.png)

**Comment by Robinlovelace** [comment]
Just checking for London. After reducing the size substantially, it loads OK locally. May make the server work hard though, but isn't that mush slower than the LSOA zones. Heads-up @nikolai-b in case this causes performance issues I can shrink it further:

![image](https://user-images.githubusercontent.com/1825120/53631147-cb053880-3c09-11e9-85c9-ccc6811444fe.png)

**Comment by Robinlovelace** [comment]
Results for WY look good to me:

![image](https://user-images.githubusercontent.com/1825120/53631472-8332e100-3c0a-11e9-8f47-ff9d72d8657a.png)

**Comment by Robinlovelace** [comment]
@nikolai-b please test this new data on the test server.

**Comment by Robinlovelace** [comment]
Update: new rnet for London now seems to work on test server: 

![image](https://user-images.githubusercontent.com/1825120/53637039-014ab400-3c1a-11e9-89e0-712f78698d02.png)

It's slow to load. Heads-up @AnnaGoodman1 here are the rules I used to reduce rnet sizes, which only show lines with 10+ trips everywhere: 

https://github.com/npct/pct-scripts/blob/f4bfaba52f9b1117ccfa660c841afc13bb7e579e/07b.3-regional-rnets.R#L48-L60

**Comment by Robinlovelace** [comment]
https://github.com/npct/pct-scripts/blob/f4bfaba52f9b1117ccfa660c841afc13bb7e579e/07b.3-regional-rnets.R#L48-L60

**Comment by AnnaGoodman1** [comment]
Sorry, what exactly did you do? I'm not sure if I'm looking at the right place, are you saying that only route segments with at least 10 cyclists in the go Dutch scenario are included (line 18 of that code)? With potentially this getting much stricter, only including 100+ cyclist segments, for very large files? (lines 48+). The latter restrictions (needing 100+ cyclists) sound too much to me, won't that give a very patchy pattern? but maybe I've misunderstood. 

 could you also update the region_stats file to explain how the rnet now works? before it was based on lines with at least 5 (10 in London) children by any mode

**Comment by Robinlovelace** [comment]
> Great, thanks but sorry, what exactly did you do? I'm not sure if I'm looking at the right place, are you saying that only route segments with at least 10 cyclists in the go Dutch scenario are included (line 18 of that code)? With potentially this getting much stricter, only including 100+ cyclist segments, for very large files? (lines 48+). The latter restrictions (needing 100+ cyclists) sound too much to me, won't that give a very patchy pattern? but maybe I've misunderstood.

Hi @AnnaGoodman1 yes that's correct. I wanted to be cautious to avoid crashing the server and only increased the minimum allowed for London after testing it and it seeming to be unusable. I think we should try different configurations. Simplification is an option but, having given it some thought, I suspect the +s of having the full route network outweigh the -s of some places being missing. Those -s are further reduced by the LSOA route network image, which gives a richer picture. 

One to discuss based on the evidence I think: please have a play with the current route network for London deployed on the test server, including increasing the slider to 90%, for one line of evidence. Suggest we catch up on Monday morning on this.

Cheers for asking the questions on this in answer to my 'heads-up' about how I've approached a tricky issue. Any suggestions, based on consideration of the tool's performance, particularly in London, very welcome.

**Comment by AnnaGoodman1** [comment]
Thanks for this @Robinlovelace . I really don't think it does work so well for London – see the picture below, it is almost all holes at 90%.  The fact that the available bits of routes mapped very well on to actual roads seems to me far less of an advantage than it would be to have much more of the route network.  I would suggest using 10 children in Go Dutch for all regions, but simplifying as needed in London - would that be an option? Or maybe up to 20 in London, one could try...

![robin](https://user-images.githubusercontent.com/11818225/53702128-817c3f80-3dfb-11e9-9d90-2bc77b4022f8.png)

**Comment by AnnaGoodman1** [comment]
another more radical option would be to have the rnets available to download but not on screen at all. I don't think that is great either, but better than such a patchy one as currently in London possibly? just a thought...

**Comment by Robinlovelace** [comment]
Hi Anna, many thanks for checking. Will think about it and try some options tomorrow. I suspect we can add a higher % for London and it will be OK, just a little slow to load...

**Comment by Robinlovelace** [comment]
One other think I considered: add the entire route network to the .geojson files where size is less of an issue. Do you think providing the files unfiltered is the best plan there? More tomorrow.

**Comment by Robinlovelace** [comment]
@AnnaGoodman1 I've re-tested the results with dutch_slc > 20 and, although the results look better (see below) it crashes the browser - it's basically unusable:

![image](https://user-images.githubusercontent.com/1825120/53723855-8d4e1d00-3e60-11e9-89f6-333e5f3808e3.png)

**Comment by Robinlovelace** [comment]
I suspect there are not many large state schools in central London.

**Comment by Robinlovelace** [comment]
Plan to push the results for 50 as a compromise (my priority was not crashing people's browsers or the server, so was deliberately risk averse in pushing these regional rnets).

Happy for this script, and the resulting rnets, to updated though: https://github.com/npct/pct-scripts/blob/master/07b.3-regional-rnets.R

**Comment by Robinlovelace** [comment]
Update: new (much simpler) seem to work fine for the big region of West Yorkshire, so happy to push this: 

![image](https://user-images.githubusercontent.com/1825120/53726817-b2de2500-3e66-11e9-8e7d-36d974502989.png)

**Comment by Robinlovelace** [comment]
90% zoom works (just about!) but is slow to load: 

![image](https://user-images.githubusercontent.com/1825120/53726900-e620b400-3e66-11e9-9d84-249fd734bc44.png)

**Comment by Robinlovelace** [comment]
Closing this issue for now - the rnets have been updated. @AnnaGoodman1 please test and feel free to re-open and change the settings/documentation. @nikolai-b please update the sha on the test server and ensure the updated data for London doesn't crash it. Many thanks all - worked more time than I was expecting on this, partly covering @mem48 (great work on building the national rnet in any case, a huge achievement).

**Comment by Robinlovelace** [comment]
Another point in favour of reducing London's school vector rnet size again, after conversation with @JDWoodcock:

It takes a long time to load after I changed it from godutch_slc >= 100 to goductch_slc >= 50, following discussion here (which may be a better forum to discuss this known issue): https://github.com/npct/pct-shiny/issues/676


As mentioned there, I think the +s of responsiveness outweigh the -s of seeing a less complete network. So happy to change back to a godutch_slc >=100, or perhaps a compromise, godutch_slc >=75. My preference is to go to back to godutch_slc >=100 *but explain clearly in the documentation that the full network can be downloaded from the geojson* (which I've set to godutch_slc >=10 in all cases) and that the 'Image' layer is more comprehensive.


Have also put these thoughts on the github issue. Thoughts from you and others will help resolve this - and let me know what you think of my suggestions (well documented less complete but more responsive online vector rnets, with more complete ones in geojson and raster tiles well-documented).

**Comment by JDWoodcock** [comment]
i would agree with @AnnaGoodman1 maybe just for download for London and show only raster

**Comment by Robinlovelace** [comment]
It could be a lot of work to make features, and associated UI, different in one place compared with another. Perhaps a better option would be to create a pop-up for London that says "Online vector route network only shows segments with Go Dutch values of 100+ due to the size of the network. See the image layer or the downloadable route network layer in GeoJSON format from the 'Region data' tab for a more complete network".

Heads-up @nikolai-b who has the skills to do this: what are your thoughts on this proposed UI change? Would a message be easy to create?

**Comment by JDWoodcock** [comment]
@RachelAldred could you have a look at London

**Comment by Robinlovelace** [comment]
At present it seems we have a compromise that hits neither objective: it's an incomplete network, and is non-responsive. A complete network cannot be done using this set-up. And removing the clickable network altogether would be inconsistent and take lots of time I don't think we have. I suspect the option outlined here https://github.com/npct/pct-shiny/issues/676#issuecomment-470038955 may be the least bad option, unless I've missed some other viable alternatives.

**Comment by RachelAldred** [comment]
I'm just on break from PCT training so having a quick look only but will try and test again more towards the end of the day when we finish if I get time. It does seem both a bit slow and still somewhat holey for London on clickable, a shame when the corresponding raster looks so nice. (We don't show clickable LSOA-level for commuting - so I don't think it would be completely inconsistent not to have it for school travel and perhaps just offer the download... but appreciate maybe it's too late to make that decision).

**Comment by JDWoodcock** [comment]
i agree would it be possible to launch all without clickable route network then revisit?

**Comment by Robinlovelace** [comment]
> i agree would it be possible to launch all without clickable route network then revisit?

That would be possible and desirable from a UI complexity perspective. Clear signposting to the vector rnet downloads would mitigate the negative of not having it. So, after pondering it more, my position on this has shifted. I support removing the clickable vector rnet layer.

Assuming there is no objections from @AnnaGoodman1 or anyone else, I'll re-open this [placeholder](https://github.com/npct/pct-shiny/issues/687) issue.

**Comment by AnnaGoodman1** [comment]
I think probably less work to keep them in, and just have a not wonderful one for London. also, even in London it does add at least a little bit of value to be able to see exactly how many cyclists are on one of the biggest routes, and not just know it is '100-199' or whatever

 But I don't feel very strongly on this, especially if someone else (@Robinlovelace  ?) is proposing to do the work of removing them. if you want to remove them you will need to change the manual (maybe) and region stats file and on the interface (definitely). unfortunately you will still have to address most of the other issues I have just raised, e.g. anonymisation

**Comment by Robinlovelace** [comment]
Pros and cons, and I don't have strong views on this either. I think we need to agree on a position. My proposal was

> go to back to godutch_slc >=100 but explain clearly in the documentation that the full network can be downloaded from the geojson (which I've set to godutch_slc >=10 in all cases) and that the 'Image' layer is more comprehensive.

Or we could leave as I set it, which is very slow. Assuming that people can live with an incomplete network (which is fine due to the downloads and the newly finished tiles IMO), I would suggest increasing the minimum `dutch_slc` value to 100, as I did before based on testing. But also happy with no change. Closing this issue for now, as the original task, to update the rnets, has been done. We need to decide a) whether we're going to remove the layer altogether (default: no, see placeholder issue) and b) whether to change the size of the deployed rnets (default: no). Those are separate issues that I think we should only create/open when we've made a collective decision.

Default as I understand it is leave it as it. I have a mild preference for making the London schools rnet smaller and faster, but have not seen sufficient support for this to act on that preference. If others support that, let me know and it can be sorted quickly. Option a) would take longer, which should also be taken into consideration.

Happy for this issue to be re-opened once a decision has been made.
